filters/include/pcl/filters/experimental/transform_filter.h

```diff
@@ -0,0 +1,126 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2020-, Open Perception
+ *
+ *  All rights reserved
+ */
+
+#pragma once
+
+#include <pcl/common/point_tests.h>
+#include <pcl/filters/filter.h>
+
+namespace pcl {
+namespace experimental {
+
+#define GET_POINT_TYPE(GridStructT) typename GridStructT::PointCloud::PointType
+
+/** \brief @b TransformFilter represents the base class for filters that performs some
+ * grid based tranformation on a point cloud, contrast to FilterIndices that performs
+ * binary point removel. The tranformation can defined by a GridStruct object passed to
+ * the class.
+ * \ingroup filters
+ */
+template <typename GridStruct, typename PointT = GET_POINT_TYPE(GridStruct)>
+class TransformFilter : public Filter<PointT> {
+protected:
+  using Filter<PointT>::filter_name_;
+  using Filter<PointT>::getClassName;
+  using Filter<PointT>::input_;
+
+  using PointCloud = typename Filter<PointT>::PointCloud;
+  using PointCloudPtr = typename PointCloud::Ptr;
+  using PointCloudConstPtr = typename PointCloud::ConstPtr;
+
+public:
+  /** \brief Empty constructor. */
+  TransformFilter()
+  {
+    if (grid_struct_.filter_name_.empty())
+      PCL_WARN("[pcl::%s] Filter name is empty\n", getClassName().c_str());
+
+    filter_name_ = grid_struct_.filter_name_;
+  }
+
+  /** \brief Destructor. */
+  ~TransformFilter() {}
+
+protected:
+  /** \brief Return const reference of the GridStruct object */
+  const GridStruct&
+  getGridStruct() const
+  {
+    return grid_struct_;
+  }
+
+  /** \brief Return reference of the GridStruct object */
+  GridStruct&
+  getGridStruct()
+  {
+    return const_cast<GridStruct&>(
+        const_cast<const TransformFilter*>(this)->getGridStruct());
```

Comment:
Isn't const cast sole role to cast away the constness? Wouldn't a static cast do the job here?

(or just write `return grid_struct` if you can)

Reply:
Done, I was overcomplicated things 😆  Now it's `return grid_struct`